scripts/migrations/2020-10-archivematica_bags_migration.py

```diff
@@ -0,0 +1,246 @@
+        client=elastic_client, index=elastic_index, size=5, query=elastic_query
+    )
+
+    with tqdm(total=document_count, file=sys.stdout) as pbar:
+        for result in results:
+            transform(result)
+            sys.exit(1)
+            pbar.update(1)
+
+
+def bag_creator(
+    workflow_s3_client, storage_s3_client, storage_client, s3_upload_bucket
+):
+    def create_bag(result):
+        document = result["_source"]
+
+        id = document["id"]
+        bucket = document["location"]["bucket"]
+        path = document["location"]["path"]
+        version = f"v{document['version']}"
+        space = document["space"]
+        external_identifier = document["info"]["externalIdentifier"]
+
+        provider = document["location"]["provider"]
+        assert provider == "amazon-s3"
+
+        path_prefix = f"s3://{bucket}/{path}"
+
+        working_id = id.replace("/", "_")
+        target_folder = "target"
+        working_folder = f"{target_folder}/{working_id}"
+        tagmanifest_name = "tagmanifest-sha256.txt"
+        archive_name = f"{working_id}.tar.gz"
+        s3_upload_prefix = "born-digital/archivematica-uuid-update"
+
+        if not os.path.exists(working_folder):
+            os.makedirs(working_folder)
+
+        prefix_patterns = ["bagit.txt", "bag-info.txt", "manifest-", "tagmanifest-"]
+
+        def _write_fetch_file():
+            def _create_fetch_line(file):
+                return f"{path_prefix}/{file['path']}\t{file['size']}\t{file['name']}\n"
+
+            files = document["files"]
+
+            fetch_file = open(f"{working_folder}/fetch.txt", "w")
+            for file in files:
+                fetch_file.write(_create_fetch_line(file))
+            fetch_file.close()
+
+        def _filter_bag_files(prefix):
+            response = storage_s3_client.list_objects_v2(
+                Bucket=bucket, Prefix=f"{path}/{version}/{prefix}"
+            )
+
+            if "Contents" in response:
+                return [content["Key"] for content in response["Contents"]]
+            else:
+                return []
+
+        def _check_expected_prefixes_exist(file_locations):
+            for prefix in prefix_patterns:
+                found_match = False
+
+                for location in file_locations:
+                    if prefix in location:
+                        found_match = True
+
+                if not found_match:
+                    raise RuntimeError(f"Missing any files matching prefix: {prefix}")
+
+        def _get_bag_files():
+            file_locations = []
+            for prefix in prefix_patterns:
+                file_locations = file_locations + _filter_bag_files(prefix)
+
+            _check_expected_prefixes_exist(file_locations)
+
+            for location in file_locations:
+                filename = location.split("/")[-1]
+                save_path = f"{working_folder}/{filename}"
+                storage_s3_client.download_file(bucket, location, save_path)
```

Comment:
Couple of suggestions:

* I would call this _get_bag_manifest_files to be really explicit that you mean things like fetch.txt and the tag manifest, not payload files.
* You can already get a list of these files from the `"tagManifest"` entry on a JSON storage manifest from the storage service.